### ISSUE TITLE
DNS cannot be resolved via TCP when strict mTLS is enabled

### ISSUE THREAD

**Issue by taylorreece** [istio/istio#19968]
**Bug description**
This seems very similar to #11658, which was marked "resolved", but I don't believe is.  Using a basic install of istio with mTLS enabled, containers cannot resolve DNS via TCP (only via UDP).  This is troublesome because DNS's behavior is to fall back to TCP if the DNS response is large (larger than 512 bytes, I think?).  For example, in a container with strict mTLS enabled, 
```
/ # nslookup -type=srv istio-example-short.reecemath.com
Server:		10.96.0.10
Address:	10.96.0.10#53

Non-authoritative answer:
istio-example-short.reecemath.com	service = 2 10 5000 istio-example2.reecemath.com.
istio-example-short.reecemath.com	service = 1 10 5000 istio-example1.reecemath.com.
```
resolves fine because it uses UDP, but
```
/ # nslookup -type=srv istio-example-long.reecemath.com
;; Truncated, retrying in TCP mode.
;; communications error to 10.96.0.10#53: connection reset
```
does not because it falls back to TCP.

**Expected behavior**
I would expect to be able to use TCP to resolve DNS, but cannot.

**Steps to reproduce the bug**
1) Download the latest istio (1.4.3 at the time of writing) and install istio with mTLS enabled
```
$ ./istioctl manifest apply --set values.global.mtls.enabled=true --set values.global.controlPlaneSecurityEnabled=true
```
2) Throw a container into istio that has mTLS injection enabled.
```
$ kubectl create ns sleep
$ kubectl label namespace sleep istio-injection=enabled
$ kubectl apply -f ../samples/sleep/sleep.yaml -n sleep
```
3) Shell into the running container
```
$ kubectl get pods -n sleep
NAME                     READY   STATUS    RESTARTS   AGE
sleep-666475687f-675t2   2/2     Running   0          67m
$ kubectl exec -it sleep-666475687f-675t2 -n sleep -- /bin/sh
Defaulting container name to sleep.
Use 'kubectl describe pod/sleep-666475687f-675t2 -n sleep' to see all of the containers in this pod.
/ # dig +short +tcp google.com
;; communications error to 10.96.0.10#53: connection reset
/ # dig +short google.com
172.217.4.78
```

Throwing part of the YAML recommended on https://github.com/istio/istio/issues/11658#issuecomment-475709922 at the cluster "fixes" the problem -- TCP can be used for DNS resolution.  You just need a `DestionationRule` to declare that you shouldn't use TLS for kube-dns.  I currently have this yaml in my cluster build pipeline now as a "workaround", but it seems like a fix within istio's installers would be more ideal.
```
apiVersion: networking.istio.io/v1alpha3
kind: DestinationRule
metadata:
  name: disable-mtls-to-dns
  namespace: kube-system
spec:
  host: "kube-dns.kube-system.svc.cluster.local"
  trafficPolicy:
    tls:
      mode: DISABLE
```

**Version**
```
$ ./istioctl version --remote
client version: 1.4.3
control plane version: 1.4.3
data plane version: 1.4.3 (2 proxies)

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.0", GitCommit:"2bd9643cee5b3b3a5ecbd3af49d09018f0773c77", GitTreeState:"clean", BuildDate:"2019-09-18T14:36:53Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.8", GitCommit:"211047e9a1922595eaa3a1127ed365e9299a6c23", GitTreeState:"clean", BuildDate:"2019-10-15T12:02:12Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```

**How was Istio installed?**
```
./istioctl manifest apply --set values.global.mtls.enabled=true --set values.global.controlPlaneSecurityEnabled=true --set values.global.mtls.auto=false
```

**Environment where bug was observed (cloud vendor, OS, etc)**
docker-desktop and GCP/GKE

**Comment by howardjohn** [comment]
We have shifted our policy to not create istio configs at install time. See https://github.com/istio/istio/issues/22095. If you want to connect to services over plaintext with strict mTLS you must explicitly declare that, we do not want to manage that for the users. If you don't want to **enforce** mTLS you may prefer auto mtls.

**Comment by howardjohn** [comment]
I didn't mean to close

**Comment by howardjohn** [comment]
I beleive this is fixed in 1.5+ as we no longer creating this config as part of the install. See https://github.com/istio/istio/issues/22095
